ctapipe/plotting/bokeh_event_viewer.py

```diff
@@ -0,0 +1,489 @@
+import numpy as np
+from bokeh.layouts import layout, column
+from bokeh.models import Select, Span
+from ctapipe.core import Component
+from ctapipe.visualization.bokeh import CameraDisplay, WaveformDisplay
+
+
+class BokehEventViewerCamera(CameraDisplay):
+    def __init__(self, event_viewer, fig=None):
+        """
+        A `ctapipe.visualization.bokeh.CameraDisplay` modified to utilise a
+        `ctapipe.core.container.DataContainer` directly.
+
+        Parameters
+        ----------
+        event_viewer : BokehEventViewer
+            The BokehEventViewer this object belongs to
+        fig : bokeh.plotting.figure
+            Figure to store the bokeh plot onto (optional)
+        """
+        self._event = None
+        self._view = 'r0'
+        self._telid = None
+        self._channel = 0
+        self._time = 0
+        super().__init__(fig=fig)
+
+        self._view_options = ['r0', 'r1', 'dl0', 'dl1', 'peakpos', 'cleaned']
+        self.w_view = None
+        self._geom_tel = None
+
+        self.event_viewer = event_viewer
+
+    def _reset(self):
+        self.reset_pixels()
+        self.event_viewer.change_time(0)
+
+    def _set_image(self):
+        e = self.event
+        v = self.view
+        t = self.telid
+        c = self.channel
+        time = self.time
+        if e:
+            tels = list(e.r0.tels_with_data)
+            if t is None:
+                t = tels[0]
+            if t not in tels:
+                raise KeyError("Telescope {} has no data".format(t))
+            if v == 'r0':
+                samples = e.r0.tel[t].waveform
+                if samples is None:
+                    self.image = None
+                else:
+                    self.image = samples[c, :, time]
+                self.fig.title.text = 'R0 Slice (T = {})'.format(time)
+            elif v == 'r1':
+                samples = e.r1.tel[t].waveform
+                if samples is None:
+                    self.image = None
+                else:
+                    self.image = samples[c, :, time]
+                self.fig.title.text = 'R1 Slice (T = {})'.format(time)
+            elif v == 'dl0':
+                samples = e.dl0.tel[t].waveform
+                if samples is None:
+                    self.image = None
+                else:
+                    self.image = samples[c, :, time]
+                self.fig.title.text = 'DL0 Slice (T = {})'.format(time)
+            elif v == 'dl1':
+                self.image = e.dl1.tel[t].image[c, :]
+                self.fig.title.text = 'DL1 Image'
+            elif v == 'peakpos':
+                peakpos = e.dl1.tel[t].peakpos
+                if peakpos is None:
+                    self.image = np.zeros(self.image.shape)
```

Comment:
For the other cases you say:
```
if samples is None:
    self.image = None
```

but for peakpos, there is an exceptional handling of this case. Could you explain the difference?